polkadot/runtime/parachains/src/paras_inherent/tests.rs

```diff
@@ -1465,6 +1482,58 @@ mod enter {
 			assert_eq!(dispatch_error, Error::<Test>::InherentOverweight.into());
 		});
 	}
+
+	#[test]
+	fn v2_descriptors_are_filtered() {
+		new_test_ext(MockGenesisConfig::default()).execute_with(|| {
+			let mut backed_and_concluding = BTreeMap::new();
+
+			for i in 0..10 {
+				backed_and_concluding.insert(i, i);
+			}
+
+			let scenario = make_inherent_data(TestConfig {
+				dispute_statements: BTreeMap::new(),
+				dispute_sessions: vec![],
+				backed_and_concluding,
+				num_validators_per_core: 5,
+				code_upgrade: None,
+				fill_claimqueue: false,
+				elastic_paras: BTreeMap::new(),
```

Comment:
we should also test with some chained candidates. to make sure that a candidate being filtered ends up filtering all of its descendants also

Reply:
The only thing to test here is filtering is happening as expected given the input. A descriptor being v2 or not does not make any difference to descendants filtering, at least at this point when there is no runtime support for v2 and everything works as before. Otherwise, this should have been already been tested when we added the code for elastic scaling MVP.

Reply:
If for example you do the filtering after the `filter_unchained_candidates` (by mistake), no tests would fail but the unchained candidates will not be filtered. And the inherent would return an error during execution.

> Otherwise, this should have been already been tested when we added the code for elastic scaling MVP.

this is being tested. but you modified the logic, added a new thing that can be filtered

Reply:
> If for example you do the filtering after the `filter_unchained_candidates` (by mistake), no tests would fail but the unchained candidates will not be filtered. And the inherent would return an error during execution.

Yeah, but the filtering is at the top of the function in this PR, why should I test something that is not implemented and will never be ?

> 
> > Otherwise, this should have been already been tested when we added the code for elastic scaling MVP.
> 
> this is being tested. but you modified the logic, added a new thing that can be filtered

Quite the opposite, the filtering itself ensures that we don't yet need any logic changes. Logic changes will come when we implement support for v2 descriptors.

Reply:
> Yeah, but the filtering is at the top of the function in this PR, why should I test something that is not implemented and will never be ?

If I add some code A that does something but also is supposed to be called before some other existing code B, I wouldn't find anything wrong with writing some test that asserts that A is indeed called before B (by verifying the output). In this case, the filtering you added must be called before the other filtering (otherwise it will output invalid inherent data).

> Quite the opposite, the filtering itself ensures that we don't yet need any logic changes. Logic changes will come when we implement support for v2 descriptors.

I meant that you modified the filtering logic. Not other logic.

Anyway, no test suite can ever test all corner cases, so it's up to you if you think it's valuable or not. It was my suggestion

Reply:
I will add a comment about the importance of doing this filtering first.

Reply:
Added comment and extended current test to have chained v2 and v1 candidates.